framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,588 @@
+     * annotations in newATM.
+     * If curAnnos is not empty, the updated set will be the LUB between
+     * curAnnos and newATM.
+     *
+     * @param curAnnos set of annotations located somewhere in a Scene.
+     * @param atf the annotated type factory of a given type system, whose
+     * type hierarchy will be used to update curAnnos.
+     * @param jaifPath identifies a Scene.
+     * @param newATM the type containing a set of annotations to be
+     * added (or "lubbed") to the Scene.
+     */
+    private static void updateAnnotationSetInScene(Set<Annotation> curAnnos,
+            AnnotatedTypeFactory atf, String jaifPath,
+            AnnotatedTypeMirror newATM) {
+        if (curAnnos != null && curAnnos.size() > 0) {
+            AnnotatedTypeMirror curAnnosATM = setOfAnnotationsToATM(
+                    curAnnos, atf, false, newATM.getUnderlyingType());
+            if (curAnnosATM != null) {
+                newATM = AnnotatedTypes.leastUpperBound(
+                        atf.getProcessingEnv(), atf, newATM, curAnnosATM);
+            }
+        }
+
+        Set<Annotation> setOfAnnos = atmToSetOfAnnotations(newATM, atf);
+        curAnnos.clear();
+        curAnnos.addAll(setOfAnnos);
+        addModifiedScene(jaifPath);
+    }
+
+    /**
+     * Returns the AField in an AScene, given an AClass and a FieldAccessNode.
+     */
+    private static AField getJaifField(AClass clazz, FieldAccessNode lhs) {
+        String fieldName = lhs.getFieldName();
+        AField field = clazz.fields.get(fieldName);
+        if (field == null) {
+            field = clazz.fields.vivify(fieldName);
+        }
+        return field;
+    }
+
+    /**
+     * Returns the AMethod in an AScene, given an AClass and a MethodTree.
+     */
+    private static AMethod getJaifMethod(AClass clazz, String methodName,
+            AScene scene) {
+        AMethod method = clazz.methods.get(methodName);
+        if (method == null) {
+            method = clazz.methods.vivify(methodName);
+        }
+        return method;
+    }
+
+    /**
+     * Returns the AClass in an AScene, given a ClassSymbol.
+     */
+    private static AClass getJaifClass(ClassSymbol classSymbol, AScene scene) {
+        String className = classSymbol.getQualifiedName().toString();
+        AClass clazz = scene.classes.get(className);
+        if (clazz == null) {
+            clazz = scene.classes.vivify(className);
+        }
+        if (className.equals("")) {
+            // TODO: Handle anonymous classes soundly, if possible
+            // (I can't think of a way to do that).
+            // Their name is equal to an empty String. For now we are ignoring them.
+            return null;
+        }
+        return clazz;
+    }
+
+
+    /**
+     * Returns true if am should not be inserted in source code,
+     * but is rather an implementation detail.
+     * I.E. {@link org.checkerframework.common.value.qual.BottomVal}.
```

Comment:
Use "e.g." for "for example".  "i.e." means "that is" -- it gives the only possibility rather than just one example.
Or, you can just use "for example" :-)